### PR TITLE
Replace trove with Apache compatible library 

### PR DESCRIPTION
**Pull request by stechu**
Replace jave premitive library Trove4J with Apache licenced HPPC.

contribute to #630

### CODE REVIEW

File: src/edu/washington/escience/myria/operator/Difference.java [hunk 1]
```diff
@@ -163,7 +161,7 @@ public void init(final ImmutableMap<String, Object> execEnvVars) throws DbExcept
       throw new DbException("Incompatible input schemas");
     }
 
-    tupleIndices = new TIntObjectHashMap<TIntList>();
+    tupleIndices = new IntObjectOpenHashMap<IntArrayList>();
```
Comment (dhalperi) [inline src/edu/washington/escience/myria/operator/Difference.java]: Would prefer `new IntObjectOpenHashMap<>()` here. `javac` will infer the correct generic type, and then there's less work to refactor in the future.

(This trick was discovered fairly late in Myria, so hasn't made it through much of the code.)

**Comment by dhalperi** [comment]
In my tests today, all 3 libraries (incl. original, actually) seem like a wash in terms of performance. So, go ahead and refactor to whatever library you prefer most :)

**Comment by dhalperi** [comment]
@stechu Want to re-open this and ping when it's actually ready for merge? Thanks.

**Comment by stechu** [comment]
@dhalperi welcome to review now.

File: src/edu/washington/escience/myria/operator/KeepAndSortOnMinValue.java [hunk 2]
```diff
@@ -182,16 +183,14 @@ public TupleBatch update(final TupleBatch tb) {
...
 
     @Override
-    public boolean execute(final int index) {
+    public void value(final int index) {
       if (TupleUtils.tupleEquals(inputTB, keyColIndices, row, uniqueTuples, keyColIndices, index)) {
         unique = false;
         Column<?> valueColumn = inputTB.getDataColumns().get(valueColIndex);
         if (shouldReplace(index, valueColumn, row)) {
           uniqueTuples.replace(valueColIndex, index, valueColumn, row);
-          replaced = true;
```
Comment (dhalperi) [inline src/edu/washington/escience/myria/operator/KeepAndSortOnMinValue.java]: @jingjingwang @stechu hmm, is this missing line a bug?

If it is, it doesn't trigger a bug in the tests...

**Comment by leventov** [comment]
@stechu see recent update of fasutil (that you haven't included into you comparison for some reason). The author claims now fastutil is the fastest lib, except the cases when Koloboke benefit from parallel layout (obj-obj, int-int, long-double maps and so on).

https://groups.google.com/forum/#!topic/fastutil/eyBRd87sqCk

**Comment by stechu** [comment]
@leventov  Thanks for the information! I am definitely interested to try it out.

**Pull request merged**